### PR TITLE
[WIP] File Management GRPC

### PR DESCRIPTION
**Pull request by AliabbasMerchant**
@YourTechBud Please have a look

### CODE REVIEW

**Comment by YourTechBud** [comment]
change the base to v0.9.0

File: proto/server.proto [hunk 1]
```diff
@@ -140,6 +140,37 @@ message SignUpRequest {
...
+message DownloadFileRequest {
+  string path = 1; // Full path, including the file name and extension
+  Meta meta = 2;
+}
+message UploadFileRequest {
+  string path = 1;
+  string name = 2;
+  bytes payload = 3;
+  int64 size = 4; // Size of the whole file in bytes
+  string hash = 5; // Hash for the whole file
```
Comment (YourTechBud) [inline proto/server.proto]: How are you planing to use the hash? If the file is too large it won't fit in the RAM. How will you hash then?

File: proto/server.proto [hunk 2]
```diff
@@ -140,6 +140,37 @@ message SignUpRequest {
...
+}
+message DeleteFileRequest {
+  string path = 1; // Full path, including the file name and extension
+  Meta meta = 2;
+}
+message DownloadFileRequest {
+  string path = 1; // Full path, including the file name and extension
+  Meta meta = 2;
+}
+message UploadFileRequest {
```
Comment (YourTechBud) [inline proto/server.proto]: Which fields will be present in the subsequent requests?

File: proto/server.proto [hunk 3]
```diff
@@ -140,6 +140,37 @@ message SignUpRequest {
...
+  string path = 1;
+  string name = 2;
+  bytes payload = 3;
+  int64 size = 4; // Size of the whole file in bytes
+  string hash = 5; // Hash for the whole file
+  Meta meta = 6;
+}
+message FilePayload {
```
Comment (YourTechBud) [inline proto/server.proto]: Which fields will be present in the subsequent payloads?

File: proto/server.proto (hunk 1 above)
Comment (AliabbasMerchant) [reply proto/server.proto]: I was thinking that we store the file, and then check the hash.

File: proto/server.proto (hunk 2 above)
Comment (AliabbasMerchant) [reply proto/server.proto]: path, name, payload, meta.
Do you want to keep a uuid also as a field?

File: proto/server.proto (hunk 3 above)
Comment (AliabbasMerchant) [reply proto/server.proto]: Only the payload.
Do you want to keep a uuid also as a field?

File: proto/server.proto (hunk 1 above)
Comment (YourTechBud) [reply proto/server.proto]: Let's add that in a future release? You can open an issue for it. Let's not do hash of now

File: proto/server.proto (hunk 2 above)
Comment (YourTechBud) [reply proto/server.proto]: only send the payload

File: proto/server.proto (hunk 3 above)
Comment (YourTechBud) [reply proto/server.proto]: nope. The payload is fine

File: proto/server.proto (hunk 2 above)
Comment (AliabbasMerchant) [reply proto/server.proto]: Ok

File: proto/server.proto (hunk 3 above)
Comment (AliabbasMerchant) [reply proto/server.proto]: Ok

**Pull request closed without merging**